src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1703 @@
+                                                              &column_handles, &kvstore_->db_));
+    kvstore_->vertex_chandle = column_handles[0];
+    kvstore_->edge_chandle = column_handles[1];
+    kvstore_->default_chandle = column_handles[2];
+  } else {
+    logging::AssertRocksDBStatus(rocksdb::TransactionDB::Open(kvstore_->options_, rocksdb::TransactionDBOptions(),
+                                                              config.disk.main_storage_directory, &kvstore_->db_));
+    logging::AssertRocksDBStatus(
+        kvstore_->db_->CreateColumnFamily(kvstore_->options_, vertexHandle, &kvstore_->vertex_chandle));
+    logging::AssertRocksDBStatus(
+        kvstore_->db_->CreateColumnFamily(kvstore_->options_, edgeHandle, &kvstore_->edge_chandle));
+  }
+}
+
+DiskStorage::~DiskStorage() {
+  durability_kvstore_->Put(lastTransactionStartTimeStamp, std::to_string(timestamp_));
+  logging::AssertRocksDBStatus(kvstore_->db_->DestroyColumnFamilyHandle(kvstore_->vertex_chandle));
+  logging::AssertRocksDBStatus(kvstore_->db_->DestroyColumnFamilyHandle(kvstore_->edge_chandle));
+  if (kvstore_->default_chandle) {
+    // We must destroy default column family handle only if it was read from existing database.
+    // https://github.com/facebook/rocksdb/issues/5006#issuecomment-1003154821
+    logging::AssertRocksDBStatus(kvstore_->db_->DestroyColumnFamilyHandle(kvstore_->default_chandle));
+  }
+  delete kvstore_->options_.comparator;
+  kvstore_->options_.comparator = nullptr;
+}
+
+DiskStorage::DiskAccessor::DiskAccessor(DiskStorage *storage, IsolationLevel isolation_level, StorageMode storage_mode)
+    : Accessor(storage, isolation_level, storage_mode), config_(storage->config_.items) {
+  rocksdb::WriteOptions write_options;
+  auto txOptions = rocksdb::TransactionOptions{.set_snapshot = true};
+  disk_transaction_ = storage->kvstore_->db_->BeginTransaction(write_options, txOptions);
+  disk_transaction_->SetReadTimestampForValidation(transaction_.start_timestamp);
+}
+
+DiskStorage::DiskAccessor::DiskAccessor(DiskAccessor &&other) noexcept
+    : Accessor(std::move(other)), config_(other.config_) {
+  other.is_transaction_active_ = false;
+  other.commit_timestamp_.reset();
+}
+
+DiskStorage::DiskAccessor::~DiskAccessor() {
+  if (is_transaction_active_) {
+    Abort();
+  }
+
+  FinalizeTransaction();
+}
+
+/// NOTE: This will create Delta object which will cause deletion of old key entry on the disk
+std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToMainMemoryCache(
+    const rocksdb::Slice &key, const rocksdb::Slice &value) {
+  auto main_storage_accessor = vertices_.access();
+
+  const std::string key_str = key.ToString();
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromKey(key_str)));
+  if (VertexExistsInCache(main_storage_accessor, gid)) {
+    return std::nullopt;
+  }
```

Comment:
This function is called by FindVertex. FindVertex already does check whether it is contained in skip list.

Reply:
I feel like this function should only do the loading, and check should be done before